### PR TITLE
stopRequest() doesn't pass on the submit button value (based on #1596)

### PR DESCRIPTION
**Pull request by Arkni**
Fixed fiddle: https://jsfiddle.net/Arkni/sofq4qbw/9/

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -1083,6 +1089,15 @@ $.extend( $.validator, {
 			$( element ).removeClass( this.settings.pendingClass );
 			if ( valid && this.pendingRequest === 0 && this.formSubmitted && this.form() ) {
 				$( this.currentForm ).submit();
+
+				// Remove the hidden input that was used as a replacement for the
+				// missing submit button. The hidden input is added by `handle()`
+				// to ensure that the value of the used submit button is passed on
+				// for scripted submits triggered by this method
+				if ( this.submitButton ) {
+					$( "input:hidden[name='" + this.submitButton.name + "']" ).remove();
```
Comment (staabm) [inline src/core.js]: Can we scope this to the current form?

Comment (Arkni) [inline src/core.js]: Fixed!

**Comment by staabm** [comment]
I guess we need a test (and maybe a fix) that the submit button is also posted when the form gets submitted via ENTER key

**Comment by Arkni** [comment]
I'm afraid we can't simulate the default behaviour of the ENTER key, we can only simulate the keyboard event (`keypress`, `keyup` & `keydown`), other than that, it needs user interaction.

Also, if you press the ENTER key on the input of the above fiddle it will send the form alongside the first submit button. So, it doesn't need fixing.

**Comment by staabm** [comment]
> I'm afraid we can't simulate the default behaviour of the ENTER key, we can only simulate the keyboard event (keypress, keyup & keydown), other than that, it needs user interaction.

hmm wouldnt a trigger of a native form submit event be enough for testing?

**Comment by staabm** [comment]
> Also, if you press the ENTER key on the input of the above fiddle it will send the form alongside the first submit button. So, it doesn't need fixing.

perfect!

**Comment by Arkni** [comment]
> hmm wouldnt a trigger of a native form submit event be enough for testing?

Nope. Doing so will only submit the form without the value of the submit button.

**Pull request merged**